tests/test_minimizers.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import division, print_function
+import unittest
+import sys
+import warnings
+
+import numpy as np
+from scipy.optimize import minimize
+
+from symfit import (
+    Variable, Parameter, Eq, Ge, Le, Lt, Gt, Ne, parameters, ModelError, Fit, Model, FitResults
+)
+from symfit.core.objectives import MinimizeModel
+from symfit.core.minimizers import BFGS
+
+
+class TestMinimize(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        np.random.seed(0)
+
+    def test_custom_objective(self):
+        """
+        Compare the result of a custom objective with the symbolic result.
+        :return:
+        """
+        # Create test data
+        xdata = np.linspace(0, 100, 25)  # From 0 to 100 in 100 steps
+        a_vec = np.random.normal(15.0, scale=2.0, size=xdata.shape)
+        b_vec = np.random.normal(100, scale=2.0, size=xdata.shape)
+        ydata = a_vec * xdata + b_vec  # Point scattered around the line 5 * x + 105
+
+        # Normal symbolic fit
+        a = Parameter(value=0, min=0.0, max=1000)
+        b = Parameter(value=0, min=0.0, max=1000)
+        x = Variable()
+        model = a * x + b
+
+        fit = Fit(model, xdata, ydata, minimizer=BFGS)
+        fit_result = fit.execute()
+
+        def f(x, a, b):
+            return a * x + b
+
+        def chi_squared(a, b):
+            return np.sum((ydata - f(xdata, a, b))**2)
+
+        fit_custom = BFGS(chi_squared, [a, b])
+        fit_custom_result = fit_custom.execute()
```

Comment:
`fit_custom = Fit(model, xdata, ydata, minimizer=BFGS(chi_squared, [a, b]))`? I probably have the signature wrong here though.

Reply:
Not sure what that would mean since it's a mix of both paradigms. Either we use chi_squared here to manually fit with BFGS, or we let Fit take care of it. or would you like to see the objective be made explicit when calling Fit? e.g. `objective=LeastSquares`

Reply:
I was thinking to initialise Fit with an instantiates minimizer. But I forgot what the point was.
In this case I do think it's also important to specify the objective.